keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,955 @@
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+Encryption at rest has been a stable Kubernetes for a long time. Every now and
+then there had been cases where, be it by improper handling or external system
+failures, the cluster encryption got into a broken state.
+
+If a single object of a resource type cannot be decrypted, listing resources of that
+type in a path prefix containing the object always fails, even if the rest of
+the resource instances is accessible.
+
+Currently, removing a resource that causes such failures is not possible.
+A cluster administrator must access etcd directly and remove the malformed data manually.
+
+This KEP proposes a way to decrease the blast radius when an undecryptable resource
+is present in a cluster, and proposes a new delete option to completely ignore
+any storage checks in order to be able to delete a failing resource by using just
+Kubernetes API.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- find a way to list objects of a resource type where at least one object cannot
+  be properly decrypted
+- provide a way to identify objects that failed decryption
+- provide an option to delete a resource independently of its contents
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- implementing system for ignoring different types of storage errors
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Introduce a new `ListOption` that would allow the caller to ignore transformation
```

Comment:
That's a good point, and I believe a controller that performs such actions should not use the option.

I was considering a controller whose job is to patch resources as they come, e.g. injecting content to configmaps/secrets etc.

Reply:
> I was considering a controller whose job is to patch resources as they come, e.g. injecting content to configmaps/secrets etc.

It will also fail in the case of a patch since the apiserver is unable to update an object that it can't read, so instead of having the error in the informer list watcher, you would have it when performing updates.

Because of that, I tend to agree with the others that we should only update the delete options and not the list ones.

Reply:
> It will also fail in the case of a patch since the apiserver is unable to update
It would, but since you're not seeing it, you don't have a reason to try to patch it.

Nevertheless, this option will go away.